src/main/groovy/wooga/gradle/github/base/internal/DefaultGithubPluginExtension.groovy

```diff
@@ -54,23 +55,25 @@ class DefaultGithubPluginExtension implements GithubPluginExtension {
     void setToken(Provider<String> token) {
         this.token.set(token)
     }
+    @Override @Internal
+    Provider<GitHub> getClientProvider() {
+        return GithubClientFactory.clientProvider(username, password, token).
+        orElse(createProvider.call({
+          throw new IOException("could not find valid credentials for github client")
```

Comment:
So to understand this. You want to fail early when the provider is requested and no credentials are set etc. What is the runtime difference without this construct? Also why not have a
```groovy
final Project project
DefaultGithubPluginExtension(Project project) {
    this.project = project
}
```
?

Reply:
This is to keep consistent with previous behavior. The `clientProvider` method returns an empty provider when there are no credentials, and previously, client creation would throw an IOException when there are no credentials.

I do agree that putting the project as an field is better, but I was thinking in keeping the current standard of using closures. I'll change that.

Reply:
I put the project in the construct, and removed the Closures.